ui/admin/app/components/form/storage-bucket/minio/index.hbs

```diff
@@ -0,0 +1,341 @@
+  <Hds::Form::Textarea::Field
+    name='description'
+    @isOptional={{true}}
+    @value={{@model.description}}
+    @isInvalid={{@model.errors.description}}
+    disabled={{form.disabled}}
+    as |F|
+  >
+    <F.Label>{{t 'form.description.label'}}</F.Label>
+    <F.HelperText>{{t 'form.description.help'}}</F.HelperText>
+    {{#if @model.errors.description}}
+      <F.Error as |E|>
+        {{#each @model.errors.description as |error|}}
+          <E.Message>{{error.message}}</E.Message>
+        {{/each}}
+      </F.Error>
+    {{/if}}
+  </Hds::Form::Textarea::Field>
+
+  {{! scope }}
+  {{#if @model.isNew}}
+    <Hds::Form::Select::Field
+      name='scope'
+      @isRequired={{true}}
+      @width='100%'
+      disabled={{form.disabled}}
+      {{on 'change' @updateScope}}
+      as |F|
+    >
+      <F.Label>{{t 'resources.storage-bucket.form.scope.label'}}</F.Label>
+      <F.HelperText>
+        {{t 'resources.storage-bucket.form.scope.help'}}
+      </F.HelperText>
+      <F.Options>
+        <option disabled hidden selected value=''>
+          {{t 'titles.choose-an-option'}}
+        </option>
+        {{#each @scopes as |scope|}}
+          <option value={{scope.model.id}}>
+            {{scope.model.displayName}}
+          </option>
+        {{/each}}
+      </F.Options>
+    </Hds::Form::Select::Field>
+  {{else}}
+    <InfoField
+      @value={{@model.scopeModel.displayName}}
+      @icon={{if @model.scopeModel.isGlobal 'globe' 'org'}}
+      disabled={{form.disabled}}
+      as |F|
+    >
+      <F.Label>{{t 'resources.storage-bucket.form.scope.label'}}</F.Label>
+      <F.HelperText>
+        {{#if @model.scopeModel.isGlobal}}
+          {{t 'resources.storage-bucket.form.scope.help_global'}}
+        {{else}}
+          {{t 'resources.storage-bucket.form.scope.help_org'}}
+        {{/if}}
+      </F.HelperText>
+    </InfoField>
+  {{/if}}
+
+  {{! Provider / plugin.name }}
+  <Hds::Form::RadioCard::Group @name='plugin_type' @alignment='center' as |G|>
+    <G.Legend>{{t 'titles.provider'}}</G.Legend>
+    <G.HelperText>{{t 'descriptions.choose-a-provider'}}</G.HelperText>
+
+    {{#each @pluginTypes as |pluginType|}}
+      <G.RadioCard
+        @value={{pluginType}}
+        @checked={{eq pluginType @model.compositeType}}
+        @disabled={{(not @model.isNew)}}
+        {{on 'input' (fn @changePluginType pluginType)}}
+        as |R|
+      >
+        <R.Label>{{t
+            (concat 'resources.storage-bucket.plugin-types.' pluginType)
+          }}</R.Label>
+        {{#if (eq pluginType 'aws')}}
+          <R.Icon @name='aws-color' />
+        {{else}}
+          <R.Icon @name='cloud-upload' />
+        {{/if}}
+      </G.RadioCard>
+    {{/each}}
+  </Hds::Form::RadioCard::Group>
+
+  {{! endpoint_url }}
+  <Hds::Form::TextInput::Field
+    name='endpoint_url'
+    @isRequired={{true}}
+    @value={{@model.endpoint_url}}
+    @isInvalid={{@model.errors.endpoint_url}}
+    disabled={{form.disabled}}
```

Comment:
The helper text says this field can't be updated later, should this field also be readonly like the name and region?

Reply:
@ZedLi 
Good catch! I am triple checking with BE. In case can't not be updated later I will make it readonly like name and region, otherwise I will change the description.

Reply:
@ZedLi 
After [checking with BE](https://hashicorp.slack.com/archives/C04N3MLQUBY/p1713224380077909) `endpoint_url` is indeed `readonly`. This change has been applied and now `endpoint_url` is `readonly` field